src/main/webapp/player/simid_player.html

```diff
@@ -53,18 +52,59 @@
           break;
         case 'survey.html':
           let adParamsObj = [
-            {'question':'Would you like a survey?',
-             'answers':['Yes', 'No', 'Not Sure']},
-            {'question':'Are you certain?',
-             'answers':['Of course', 'Not Really', 'Well...']},
-            {'question':'What animals do you like?',
-             'answers':['Cats', 'Dogs', 'Other']}
+            {
+              'question': 'Would you like a survey?',
+              'answers': ['Yes', 'No', 'Not Sure']
+            },
+            {
+              'question': 'Are you certain?',
+              'answers': ['Of course', 'Not Really', 'Well...']
+            },
+            {
+              'question': 'What animals do you like?',
+              'answers': ['Cats', 'Dogs', 'Other']
+            }
           ];
           adParams = JSON.stringify(adParamsObj);
           break;
       }
       document.getElementById('ad_params').value = adParams;
     }
+
+    window.getLocationPermission = () => {
+      if (navigator.geolocation) {
+        navigator.geolocation.getCurrentPosition(function (position) {
+          const pos = {
+            lat: position.coords.latitude,
+            lng: position.coords.longitude
+          };
+          let adParamObject = document.getElementById('ad_params').value;
+          let adParams = JSON.parse(adParamObject);
+          adParams.userCoordinates = pos;
+          adParamObject = JSON.stringify(adParams);
+          document.getElementById('ad_params').value = adParamObject;
+        }, (error) => { //callback function that takes a GeolocationPositionError 
+          handleLocationError(error, true);
+        });
+      } else {
+        // Browser doesn't support Geolocation
+        handleLocationError(error, false);
+      }
+    }
+
+    window.handleLocationError = (error, geolocationAllowed) => {
```

Comment:
Since this is only being called from inside this module, you shouldn't have to add it to window

Reply:
fixed!